client/devpi/use.py

```diff
@@ -403,35 +398,42 @@ def main(hub, args=None):
     else:
         hub.line("no server: type 'devpi use URL' with a URL "
                  "pointing to a server or directly to an index.")
-    if current.venvdir:
-        hub.info("venv for install command: %s" % current.venvdir)
-    #else:
-    #    hub.line("no current install venv set")
+
+    venvdir = hub.venv
+    if venvdir:
+        hub.info("venv for install/set commands: %s" % venvdir)
+
     settrusted = hub.args.settrusted == 'yes'
     if hub.args.always_setcfg:
         always_setcfg = hub.args.always_setcfg == "yes"
-        hub.current.reconfigure(dict(always_setcfg=always_setcfg,
+        current.reconfigure(dict(always_setcfg=always_setcfg,
                                      settrusted=settrusted))
-    if hub.args.setcfg or hub.current.always_setcfg:
+    set_cfgs = []
+    if hub.args.setcfg or current.always_setcfg:
         if not hub.current.index:
             hub.error("no index configured: cannot set pip/easy_install index")
         else:
-            indexserver = hub.current.simpleindex_auth
-            searchindexserver = hub.current.searchindex_auth
-            DistutilsCfg().write_indexserver(indexserver)
-            PipCfg().write_indexserver(indexserver)
-            PipCfg().write_searchindexserver(searchindexserver)
-            BuildoutCfg().write_indexserver(indexserver)
+            indexserver = current.simpleindex_auth
+            searchindexserver = current.searchindex_auth
+            if venvdir:
+                hub.line("only setting venv pip cfg, no global configuration changed")
+            else:
+                for cfg in DistutilsCfg(), BuildoutCfg():
+                    cfg.write_indexserver(indexserver)
+                    set_cfgs.append(cfg)
+
+            pipcfg = PipCfg(venv=venvdir)
+            pipcfg.write_indexserver(indexserver)
+            pipcfg.write_searchindexserver(searchindexserver)
             if settrusted or hub.current.settrusted:
-                PipCfg().write_trustedhost(indexserver)
+                pipcfg.write_trustedhost(indexserver)
             else:
-                PipCfg().clear_trustedhost(indexserver)
+                pipcfg.clear_trustedhost(indexserver)
 
-    show_one_conf(hub, DistutilsCfg())
-    show_one_conf(hub, PipCfg())
-    show_one_conf(hub, BuildoutCfg())
-    hub.line("always-set-cfg: %s" % ("yes" if hub.current.always_setcfg else
-                                     "no"))
+            set_cfgs.append(pipcfg)
+    for cfg in set_cfgs:
```

Comment:
I just noticed, that the current config settings aren't shown anymore. Only when they are changed. I think that should be fixed.
Before one got output like this from ``devpi use``:
```
$ devpi use /root/pypi
current devpi index: http://localhost:3141/root/pypi (logged in as emilie)
~/.pydistutils.cfg     : http://localhost:4040/alice/dev/+simple/
~/.pip/pip.conf        : http://localhost:4040/alice/dev/+simple/
~/.buildout/default.cfg: http://localhost:4040/alice/dev/+simple/
always-set-cfg: no
```
Now the three lines starting with ``~`` are missing.

Reply:
Sure, that's certainly an unintentional change, I'll try to look into it soon.